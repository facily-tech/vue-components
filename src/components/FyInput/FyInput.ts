/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';

const base = Vue.extend({ mixins: [VTextField] });

type options = InstanceType<typeof base>;

export default base.extend<options>().extend({
  name: 'fy-input',
  computed: {
    class() {
      return {
        ...VTextField.options.computed.classes.call(this),
        'fy-input': true,
      };
    },
  },
});
