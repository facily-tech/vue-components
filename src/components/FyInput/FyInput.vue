<template>
  <v-text-field v-bind="$attrs" v-on="$listeners" dense outlined :data-test="dataTest">
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </v-text-field>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FyInput',
  props: {
    dataTest: {
      type: String,
      default: 'input',
    },
  },
});
</script>
