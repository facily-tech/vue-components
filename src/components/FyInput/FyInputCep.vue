<template>
  <div>
    <fy-input
      :value="text"
      @input="update"
      v-mask="mask"
      :rules="cepRule"
      v-bind.sync="$props"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';

const BaseVTextField = Vue.extend({ mixins: [VTextField] });

interface options extends InstanceType<typeof BaseVTextField> {
  value: string;
}

export default BaseVTextField.extend<options>().extend({
  name: 'fy-input-cep',

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      cepRule: [
        (value: string): boolean | string => {
          if (value && value !== '')
            return value.length === 9 || 'O campo Cep deve conter 8 n√∫meros';

          return false;
        },
      ],
      mask: '#####-###',
    };
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
