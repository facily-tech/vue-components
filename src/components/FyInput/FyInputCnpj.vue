<template>
  <div>
    <fy-input
      :value="text"
      @input="update"
      v-mask="mask"
      :rules="cnpjRule"
      v-bind.sync="$props"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';

const BaseVTextField = Vue.extend({ mixins: [VTextField] });

interface options extends InstanceType<typeof BaseVTextField> {
  value: string;
}

export default BaseVTextField.extend<options>().extend({
  name: 'fy-input-cnpj',

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      cnpjRule: [
        (value: string): boolean | string => {
          if (value && value !== '')
            return value.length === 18 || 'O campo Cnpj deve conter 14 n√∫meros';

          return false;
        },
      ],
      mask: '##.###.###/####-##',
    };
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
