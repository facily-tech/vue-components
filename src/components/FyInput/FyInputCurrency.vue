<template>
  <div>
    <fy-input
      :value="text"
      @input="update"
      v-money="money"
      v-bind.sync="$props"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';
import { VMoney } from 'v-money';

const BaseVTextField = Vue.extend({ mixins: [VTextField] });

interface options extends InstanceType<typeof BaseVTextField> {
  value: string;
  type: string;
  decimal: string;
  thousands: string;
  precision: number;
}

export default BaseVTextField.extend<options>().extend({
  name: 'fy-input-currency',

  directives: { money: VMoney },

  props: {
    value: {
      tyupe: String,
      default: '0',
    },
    type: {
      type: String,
      default: 'R$ ',
    },
    decimal: {
      type: String,
      default: ',',
    },
    thousands: {
      type: String,
      default: '.',
    },
    precision: {
      type: Number,
      default: 2,
    },
  },
  data() {
    return {
      text: this.value ? this.value : '0',
      money: {
        decimal: ',',
        thousands: '.',
        prefix: this.type,
        precision: 2,
        masked: false /* doesn't work with directive */,
      },
    };
  },
  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
