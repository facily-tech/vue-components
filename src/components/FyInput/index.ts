export { default as FyInput } from './FyInput';
export { default as FyInputCep } from './FyInputCep.vue';
export { default as FyInputCnpj } from './FyInputCnpj.vue';
export { default as FyInputCpf } from './FyInputCpf.vue';
export { default as FyInputCurrency } from './FyInputCurrency.vue';
export { default as FyInputDate } from './FyInputDate.vue';
export { default as FyInputDateTime } from './FyInputDateTime.vue';
export { default as FyInputPhone } from './FyInputPhone.vue';
export { default as FyInputTime } from './FyInputTime.vue';
