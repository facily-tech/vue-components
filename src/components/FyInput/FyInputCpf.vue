<template>
  <fy-input
    :value="text"
    @input="update"
    v-mask="mask"
    :rules="cpfRule"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </fy-input>
</template>

<script lang="ts">
import Vue from 'vue';
import { VueMaskDirective } from 'v-mask';
import FyInput from './FyInput.vue';

export default Vue.extend({
  name: 'fy-input-cpf',

  directives: { mask: VueMaskDirective },

  components: {
    FyInput,
  },

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      cpfRule: [
        (value: string): boolean | string =>
          value.length === 14 || 'O campo Cpf deve conter 11 n√∫meros',
      ],
      mask: '###.###.###-##',
    };
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
