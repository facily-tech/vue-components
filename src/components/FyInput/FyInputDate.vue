<template>
  <fy-input
    :value="text"
    @input="update"
    v-mask="'##/##/####'"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  />
</template>
<script lang="ts">
import Vue from 'vue';
import { VueMaskDirective } from 'v-mask';
import FyInput from './FyInput.vue';
import moment from 'moment';

export default Vue.extend({
  name: 'fy-input-date',

  directives: { mask: VueMaskDirective },

  components: {
    FyInput,
  },

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      mask: '##/##/####',
    };
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },

  methods: {
    getStringToDate(str: string): string {
      const date = moment(str, 'DD/MM/YYYY');
      return date.format();
    },

    getDateToString(date: string): string {
      if (!date) return '';
      const nDate = moment(date);
      return nDate.format('DD/MM/YYYY');
    },

    update(value: string): void {
      const formatDate = this.getStringToDate(value);
      this.$emit('input', formatDate);
    },
  },
});
</script>
