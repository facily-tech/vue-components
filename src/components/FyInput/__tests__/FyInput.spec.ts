import FyInput from '../FyInput';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

describe('FyInput.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInput>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInput, {
        mocks: {
          $vuetify: {
            icons: {},
            rtl: false,
          },
        },
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
