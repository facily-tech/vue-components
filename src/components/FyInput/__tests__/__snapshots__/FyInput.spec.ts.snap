// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInput.ts should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">â€‹</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input id="input-2" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
